src/mlpack/methods/hmm/hmm_impl.hpp

```diff
@@ -570,15 +593,26 @@ void HMM<Distribution>::Forward(const arma::mat& dataSeq,
 
   arma::mat logTrans = trans(log(transition));
 
+  // Define a variable to store the value of log-probability for dataSeq.
+  arma::mat logProbs(dataSeq.n_cols, transition.n_rows);
+
+  // Save the values of log-probability to logProbs.
+  for (size_t i = 0; i < transition.n_rows; i++)
+  {
+    // Define alias of desired column.
+    arma::vec alias(logProbs.colptr(i), logProbs.n_rows, false, true);
+    // Use advanced constructor for using logProbs directly.
+    emission[i].LogProbability(dataSeq, alias);
+  }
```

Comment:
Instead of computing `logProbs` here, can we compute it in the functions that call `Forward()` and `Backward()`?  For instance, `LogEstimate()` calls `Forward()` then `Backward()`, so it would be better to compute `logProbs` there then pass it to both `Forward()` and `Backward()`. :+1:

Reply:
Strange, I made the changes, but that didn't work out, it slowed down again 

![Screenshot from 2019-11-06 00-07-05](https://user-images.githubusercontent.com/21286134/68235516-62d4e500-0029-11ea-9c9f-aa0d5bc28c89.png)